contents/tutorials/test-frontend-feature-flags.md

```diff
@@ -14,32 +14,80 @@ Combining both testing and feature flags can be a bit tricky. Tests generally ch
 
-## Creating a React app
+## Creating a Vite-React app and setup Vitest
 
-First, create a basic React app. Make sure [Node is installed](https://nodejs.dev/en/learn/how-to-install-nodejs/) then run `npx create-react-app` with the name of your app. We named ours `flag-test`.
+First, ensure [Node.js is installed](https://nodejs.dev/en/learn/how-to-install-nodejs/) (version 18.0 or newer). Then create a new React app with Vite: We named ours `flag-test`.
 
 ```bash
-npx create-react-app flag-test
+npm create vite@latest flag-test -- --template react
 ```
 
-After creating the app, in the newly created `flag-test` folder, a test is automatically created in `src/App.test.js` that looks like this:
+After creating the app, in the newly created `flag-test` folder, but by default the test framework isn't installed at the project yet, so let's add it.
+
+```bash
+cd flag-test
+npm add -D vitest @testing-library/react @testing-library/jest-dom
+```
+
+Since we're about to test `components` directly, we must create a file called `vitest.setup.js` at our project root importing the required dependency:
 
 ```js
-// src/App.test.js
-import { render, screen } from '@testing-library/react';
+// ./vitest.setup.js
+import '@testing-library/jest-dom'
+```
+
+With the dependency installed at our project, the next step is to let the `vite.config.js` know the new testing setup configurations by adding the `test` property:
+
+```js
+// ./vite.config.js
+import { defineConfig } from 'vite'
+import react from '@vitejs/plugin-react'
+
+export default defineConfig({ 
+  plugins: [react()],
+  test: {
+    globals: true,
+    environment: 'jsdom',
+    setupFiles: './vitest.setup.js'
+  },
+})
+```
+
+Now, we're ready to set up our first test. Since the **vite-react** already provided a sample component `src/App.jsx`, we can create a new file called `src/App.test.jsx` and start with this boilerplate:
+
+
+```js
+// ./src/App.test.js
+import { expect, test } from 'vitest'
+import { render, screen } from "@testing-library/react";
 import App from './App';
 
-test('renders learn react link', () => {
+test('renders learn link', () => {
   render(<App />);
-  const linkElement = screen.getByText(/learn react/i);
+  const linkElement = screen.getByText(/learn/i);
   expect(linkElement).toBeInTheDocument();
 });
 ```
 
-This test uses Jest, a popular JavaScript testing library. It passes when all the default code is in place and is a test to build on as we build out our app.
+This test uses **Vitest**, a popular JavaScript testing library. It passes when all the default code is in place and is a test to build on as we build out our app.
+
+To run the tests, update your `package.json` with the new script:
+
+```json
+{
+  "scripts": {
+    "dev": "vite",
+    "build": "vite build",
+    "lint": "eslint .",
+    "preview": "vite preview",
+    "test": "vitest"
+  }
+}
+```
+
+With everything settled up, we can finally run our tests: 
 
 ```bash
-cd flag-test
 npm test
```

Comment:
Seems we need to add jsdom before this point too?
![CleanShot 2025-03-27 at 09 45 43@2x](https://github.com/user-attachments/assets/c4f364af-5e4f-4eec-9bd8-3519f177e2bc)

Reply:
Fixed